core/src/cloud/api.ts

```
@@ -323,15 +361,95 @@ export class CloudApi {
+    }
+
+    return project
+  }
+
+  async getProjectByName(projectName: string): Promise<CloudProject | undefined> {
+    let response: ListProjectsResponse
+
+    try {
+      response = await this.get<ListProjectsResponse>(`/projects`)
+    } catch (err) {
+      this.log.debug(`Attempt to list all projects failed with ${err}`)
+      throw err
+    }
+
+    if (response.status === "error") {
+      this.log.debug(`Attempt to retrieve projects failed with ${response.error}`)
+      throw new EnterpriseApiError(`Failed to retrieve projects for the organization`, {})
+    }
+
+    let projects: ListProjectsResponse["data"] = response.data
+    let project: ListProjectsResponse["data"][0] | undefined
+
+    project = projects.find((p) => p.name === projectName)
+
+    if (!project) {
+      return undefined
+    }
+
+    return toCloudProject(project)
+  }
+
+  async getOrCreateProject(projectName: string): Promise<CloudProject> {
+    let project: CloudProject | undefined = await this.getProjectByName(projectName)
+
+    if (!project) {
+      project = await this.createProject(projectName)
+    }
+
+    // This is necessary to internally configure the project for this instance
+    this._project = project
+    this.projectId = project.uid
+
+    return project
+  }
+
+  async createProject(projectName: string): Promise<CloudProject> {
+    let response: CreateProjectsForRepoResponse
+
+    try {
+      const createRequest = {
+        name: projectName,
+        repositoryUrl: "",
+        relativeProjectRootPath: "",
+        importFromVcsProvider: false,
+      }
+
+      response = await this.post<CreateProjectsForRepoResponse>(`/projects/`, {
+        body: createRequest,
+      })
+    } catch (err) {
+      this.log.debug(`Create project request failed with error, ${err}`)
+      throw err
+    }
+
+    if (response.status === "error" || (response.status === "success" && response.data.length !== 1)) {
```

Only check: `if (response.data.length !== 1` and log an informative and actionable error.

---

Where we catch the error in the Garden class we need to check if it's a "duplicate project error", and if so, fail hard with, again, an actionable message for the user. 

This is what happened, this is what you need to do.

---

Opted for a warning message instead and continuing without a project configured since that's how other errors are handled when we cannot use a project.